frontend/packages/knative-plugin/src/utils/create-knative-utils.ts

```diff
@@ -33,11 +33,13 @@ export const getKnativeServiceDepResource = (
     project: { name: namespace },
     serverless: { scaling },
     limits,
-    route: { unknownTargetPort, create },
+    route: { unknownTargetPort, create, targetPort },
     labels,
     image: { tag: imageTag },
   } = formData;
-  const contTargetPort: number = parseInt(unknownTargetPort, 10);
+  const contTargetPort = targetPort
+    ? parseInt(targetPort.split('-')[0], 10)
```

Comment:
Port with a dash? What is the value of `targetPort`?

Reply:
yeah we are setting it with protocol as `8080-tcp` , as Route model takes in same format. In case of knative it takes Port as number so needed to split
```
port:
    targetPort: 8080-tcp
```

can check here https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx#L72

Reply:
Ah gotcha... was not doing a review just saw the oddity and asked the question... makes sense.